src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component;
+use ILIAS\UI\Implementation\Component\Modal\LightboxImagePage;
+
+/**
+ * Interface Factory
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      An Interruptive Modal disrupts the user in critical situation,
+     *      forcing him or her to focus on the task at hand.
+	 *   composition: >
+	 *      The Modal states why this situation needs attention and may point out consequences.
+	 *   effect: >
+	 *      All controls of the original context are inaccessible until the Modal is completed.
+	 *      Upon completion the user returns to the original context.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *        Due to the heavily disruptive nature of this type of Modal it MUST be
+	 *        restricted to critical situations (e.g. loss of data).
+	 *     2: >
+	 *        All delete-actions are critical situations and SHOULD be implemented as an Interruptive Modal.
```

Comment:
There has been some discussion, whether this rule is to harsh. I propose to add some clarification: "Exceptions are possible if items from lists in forms are to be deleted or if the modal would heavily disrupt the workflow."